rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/rabbitmq/TbRabbitMqNodeTest.java

```diff
@@ -0,0 +1,221 @@
+        given(ctxMock.isExternalNodeForceAck()).willReturn(false);
+        mockOnInit();
+        ListeningExecutor listeningExecutor = mock(ListeningExecutor.class);
+        given(ctxMock.getExternalCallExecutor()).willReturn(listeningExecutor);
+        String errorMsg = "Something went wrong";
+        ListenableFuture<TbMsg> failedFuture = Futures.immediateFailedFuture(new RuntimeException(errorMsg));
+        willReturn(failedFuture).given(listeningExecutor).executeAsync(any(Callable.class));
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+        TbMsgMetaData metaData = new TbMsgMetaData();
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, DEVICE_ID, metaData, TbMsg.EMPTY_JSON_OBJECT);
+        node.onMsg(ctxMock, msg);
+
+        then(ctxMock).should(never()).ack(any(TbMsg.class));
+        ArgumentCaptor<TbMsg> actualMsg = ArgumentCaptor.forClass(TbMsg.class);
+        ArgumentCaptor<Throwable> throwable = ArgumentCaptor.forClass(Throwable.class);
+        then(ctxMock).should().tellFailure(actualMsg.capture(), throwable.capture());
+        metaData.putValue("error", RuntimeException.class + ": " + errorMsg);
+        TbMsg expectedMsg = TbMsg.transformMsgMetadata(msg, metaData);
+        assertThat(actualMsg.getValue()).usingRecursiveComparison().ignoringFields("ctx").isEqualTo(expectedMsg);
+        assertThat(throwable.getValue()).isInstanceOf(RuntimeException.class).hasMessage(errorMsg);
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void givenAMQPBasicPropertiesName_whenConvert_thenReturnAMQPBasicProperties(String name, AMQP.BasicProperties expectedBasicProperties) throws TbNodeException {
+        AMQP.BasicProperties actualBasicProperties = TbRabbitMqNode.convert(name);
+        assertThat(actualBasicProperties).isEqualTo(expectedBasicProperties);
+    }
+
+    private static Stream<Arguments> givenAMQPBasicPropertiesName_whenConvert_thenReturnAMQPBasicProperties() {
+        return Stream.of(
+                Arguments.of("BASIC", MessageProperties.BASIC),
+                Arguments.of("TEXT_PLAIN", MessageProperties.TEXT_PLAIN),
+                Arguments.of("MINIMAL_BASIC", MessageProperties.MINIMAL_BASIC),
+                Arguments.of("MINIMAL_PERSISTENT_BASIC", MessageProperties.MINIMAL_PERSISTENT_BASIC),
+                Arguments.of("PERSISTENT_BASIC", MessageProperties.PERSISTENT_BASIC),
+                Arguments.of("PERSISTENT_TEXT_PLAIN", MessageProperties.PERSISTENT_TEXT_PLAIN)
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"Basic", "TEXT_plain", "minimal basic", "", "    "})
+    public void givenUndefinedProperties_whenConvert_thenThrowsException(String name) {
+        assertThatThrownBy(() -> TbRabbitMqNode.convert(name))
+                .isInstanceOf(TbNodeException.class)
+                .hasMessage("Message Properties: '" + name + "' is undefined!");
```

Comment:
Please create a str value of supported properties, and update the error messages in the same way as we do this in other places.